pkg/operation/botanist/component/logging/event_logger.go

```diff
@@ -0,0 +1,364 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logging
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	gutil "github.com/gardener/gardener/pkg/utils/gardener"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+	secrets "github.com/gardener/gardener/pkg/utils/secrets"
+	"github.com/gardener/gardener/pkg/utils/secrets/manager"
+	secretsmanager "github.com/gardener/gardener/pkg/utils/secrets/manager"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	eventLoggerName                = "event-logger"
+	eventLoggerManagedResourceName = "shoot-event-logger"
+	eventLoggerRoleName            = "gardener.cloud:logging:event-logger"
+)
+
+// Interface contains functions for a shoot event-logger deployer.
+type Interface interface {
+	component.Deployer
+}
+
+type eventLogger struct {
+	client           client.Client
+	namespace        string
+	imageEventLogger string
+	secretsManager   secretsmanager.Interface
+}
+
+// NewEventLogger creates a new instance of event-logger for the shoot event logging.
+func NewEventLogger(
+	client client.Client,
+	namespace string,
+	secretsManager manager.Interface,
+	imageEventLogger string,
+) (Interface, error) {
```

Comment:
The `Interface` type is not needed here. You could use `component.Deployer` right away.